gcbdr/backupplan/main.tf

```diff
@@ -0,0 +1,69 @@
+/**
+* Copyright 2024 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+# [START backupdr_create_backupvault]
+
+resource "google_backup_dr_backup_vault" "default" {
+  provider                                   = google-beta
+  location                                   = "us-central1"
+  backup_vault_id                            = "bv-1"
+  description                                = "This is a second backup vault built by Terraform."
+  backup_minimum_enforced_retention_duration = "100000s"
+
+  labels = {
+    foo = "bar1"
+    bar = "baz1"
+  }
+
+  annotations = {
+    annotations1 = "bar1"
+    annotations2 = "baz1"
+  }
+
+  force_update  = "true"
+  force_delete  = "true"
+  allow_missing = "true"
+}
+
+# [END backupdr_create_backupvault]
+
+# [START backupdr_create_backupplan]
+
+# Before creating a backup plan, you need to create backup vault (google_backup_dr_backup_vault).
```

Comment:
Instead of having comments like this, add a region tag around the earlier `google_backup_dr_backup_vault` definition, and include that in your documentation before the backup_plan. This also allows your users to see both in context. (Additionally, since your plan references your vault, this resource cannot be provisioned on it's own without the associated dependency, so both should be wrapped in region tags and displayed on your documentation page.

Reply:
We have separate pagefor Backupvault. And user flow of documentation starts with BV and takes to BP. We mention BV terraform code in BV page, mentioning in BP page as well, looks repetitive?

Reply:
Also, currently datasource support is not present for backupvault, we will ensure that, we update code to use datasource instead of resource creation multiple times in dependant resources